pkg/agent/cniserver/server.go

```diff
@@ -403,6 +406,10 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (
 		klog.Errorf("Failed to configure container %s interface: %v", cniConfig.ContainerId, err)
 		return s.configInterfaceFailureResponse(err), nil
 	}
+
+	// Notify the Pod update event to required components.
+	s.podUpdates <- v1beta1.PodReference{Name: podName, Namespace: podNamespace}
```

Comment:
so as a follow-up to your comment there: https://github.com/vmware-tanzu/antrea/issues/197#issuecomment-565000505

we are assuming that it takes less time to 1) notify on the channel + reconcile rules + install flows than it takes for 2) kubelet to receive the response from the CNI and create the workload container

If 1) is very fast, wouldn't it be acceptable to just block until we know it has completed before returning from CNI Add? This way we would have a guarantee that known network policies are enforced before the workload container is created. I know that we don't have a notification mechanism to determine when flows have been installed, but that's something we may have in the future. The only issue I see is we need to know the OF port for NP flows, which means we cannot start installing flows until we have added the new port to the vSwitch.

Reply:
Even we block there there is no guarantee Controller already computes and pushes all policies applied to the Pod. As most event processing is already async, I prefer to keep the async behaviors here too.

Reply:
Agree @jianjuns. Also, as I explained in https://github.com/vmware-tanzu/antrea/pull/222#discussion_r358079265, it's not even guaranteed antrea-controller will receive the network policy and handle it first unless we sort events of different resources which makes it more complicated. I think the most important thing is whether it's worth to do it if it makes no difference to users.